File: 0062-unique-paths/0062-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        currow = [0] * (n + 1)
        dummy = [0] * n
        currow[-1] = 1
        for i in range(m - 1,-1, -1):
            for j in range(n-1, -1, -1):
                currow[j] = currow[j + 1] + dummy[j]
            dummy = currow
            currow = [0] * (n + 1)
        return dummy[0]

File: 0062-unique-paths/0062-unique-paths.js
/**
 * @param {number} m
 * @param {number} n
 * @return {number}
 */
var uniquePaths = function(m, n) {
    let currow = new Array(n + 1).fill(0);
    let dummy = new Array(n + 1).fill(0);
    currow[currow.length - 1] = 1;
    for(let i = m - 1; i > -1; i--){
        for(let j = n - 1; j > -1; j--){
            currow[j] = currow[j + 1] + dummy[j];
        }
        let hold = dummy;
        dummy = currow;
        currow = hold.fill(0);
    }
    return dummy[0];
};